_posts/2017-11-21-Installing-Jenkins-with-PowerShell-DSC.md

```diff
@@ -0,0 +1,745 @@
+---
+layout: post
+title: Installing Jenkins with PowerShell DSC
+author: Rinorragi
+date: 2017-11-21 13:20:00 +0200
+excerpt: How to use PowerShell DSC in installing your continuous integration server.
+categories: 
+- Episerver
+tags: 
+- Episerver 
+- DOTNET 
+- DevOps
+- PowerShell
+- Jenkins
+---
+I have been struggling to find a good reference about how to setup a Jenkins environment in Microsoft environment with automatic installation script. So I decided to write a blog post about it. I also want to write about how to do continuous integration with Episerver DXC but felt that I need to first tell how to setup the Jenkins. Our aim is to put up a Jenkins server that can build .NET FrameWork MVC application and front end. If you are not a reader type of a person, then [here](https://github.com/solita/powershell-dsc-jenkins) is a GitHub link to example scripts.
+
+## About the PowerShell DSC
+
+You might not be familiar with the PowerShell DSC and it comes from Desired Stage Configuration. It has been there now for years still in my experience most .NET developers are unfamiliar with it. DSC exists for various reasons:
+
+* Make scripting less complex
+* Make scripting to look the same for smaller learning curve 
+* Make scripting pieces to be more reusable
+* Make installation scripts to be idempotent (repeatable)
+
+The DSC is all about the setting the state of a machine to be certain. Most used example is to make sure that a service is running or that specific file is found on given location. If you dig further on to this world you will find concepts of pull and push servers that would help you to set a farm of machines into certain state. We will not use those but we run the script locally with the help of LCM which is "local configuration manager". If you are looking for basics of PowerShell Desired State Configuration then this [blog](https://red-gate.com/simple-talk/sysadmin/powershell/powershell-desired-state-configuration-the-basics/) was a well-written one.
+
+## DSC resources 
+
+Before we start the configuration we need to talk about DSC resources which are libraries for DSC. Resources provide you functionality for DSC. For our purpose we are needing at least one that helps us to grab software from [Chocolatey](https://chocolatey.org/) (a package repository for windows). These resources would normally be where you start the script from (which might take you back to push and pull servers). In this scenario I just install them locally. [Here](https://github.com/solita/powershell-dsc-jenkins/blob/master/install-modules.ps1) is what my Install-Modules.ps1 script has inside.
```

Comment:
Joko Before we start the configuration _work_
tai Before we start configuring,

Jos kaikki DSC-resurssit ovat DSC-kirjastoja, niin which-sana on virheellinen, pitäisi olla esim. 
... about DSC resources. DSC resources are libraries that provide DSC functinality.

For our purpose, we *need* at least one *library*